src/main/java/ch/uzh/ifi/hase/soprafs24/service/GameService.java

```diff
@@ -51,9 +51,13 @@ public Game createGame(GameParameters gameParameters) {
         Game newGame = new Game(gameParameters, host);
         Game game = inMemoryGameRepository.save(newGame);
         addPlayerToGame(game, host);
-        setPlaylistNameAndURL(game);
-        return inMemoryGameRepository.save(newGame);
+        int playlistLength = setPlaylistNameAndURL(game);
 
+        // if playlist is too short, reduce number of sets
+        if (game.getGameParameters().getNumOfSets() > playlistLength) {
+            game.getGameParameters().setNumOfSets(playlistLength);
+        }
+        return inMemoryGameRepository.save(game);
```

Comment:
would it be possible to do either of the following? IMO, it seems more logical to get such options
 - duplicate songs to meet the numberOfSets?
 - (or) throw warning before starting when unmet playlist is set?